docs/workflow-and-job.md

```diff
@@ -10,151 +10,222 @@ It can execute anything that can be executed from the Python code, for example:
-
-```python
-hello_world_workflow.run()
-hello_world_workflow.run_job('hello_world')
-```
+## Job
 
-Finally, you can build and deploy created workflow to Cloud Composer:
+A job is just an object that has a unique `id` and implements the `run` method.
 
-```shell script
-bgf build;bgf deploy
-```
+The `id` parameter is a string that should be a valid Python variable name. For example - `'my_example_job'`, `'MY_EXAMPLE_JOB'`, `'job1234'`.
 
-### Job
+The run method takes a single argument - `runtime`. The `runtime` parameter is a date-time in the form of a string. 
+You can find more information about `runtime` and scheduling [in the chapter below](#workflow-scheduling-options).
 
-Job is just an object that has a unique `id` and implements the `run` method.
+There are 2 additional parameters, that a job can supply - `retry_count` and `retry_pause_sec`. The `retry_count` parameter
+determines how many times a job will be retried (in case of a failure). The `retry_pause_sec` says how long the pause between retries should be.
 
+[`./docs_examples/workflow_and_job/retriable_job.py`](./docs_examples/workflow_and_job/retriable_job.py)
 ```python
-class SimpleJob:
+class SimpleRetriableJob:
     def __init__(self, id):
         self.id = id
+        self.retry_count = 20
+        self.retry_pause_sec = 100
 
     def run(self, runtime):
         print(runtime)
-
-job = SimpleJob('my_simple_job')
 ```
 
-The `runtime` parameter represents a date and time of job execution. Let us say that your workflow runs every day at 7 am,
-starting from 2020-01-01. Then, the `run` method will be executed for each job with `runtime` 
-equals `"2020-01-01 07:00:00"`, `"2020-01-02 07:00:00"`, `"2020-01-03 07:00:00"` and so on.
-
-The `runtime` parameters is a string. It will be formatted as either `YYYY-MM-DD` or `YYYY-MM-DD hh-mm-ss`.
-It depends on the `Workflow` setup. By default, it will be date only - `YYYY-MM-DD`.
+## Workflow
 
-When you run a single job through the workflow class, without providing the `runtime` parameter, the `Workflow` class
-will pass the current date time as default.
+The `Workflow` class takes 2 main parameters - `workflow_id` and `definition`.
 
-```python
-# Let's assume that now == '2020-01-02 01:11:00'
-simple_workflow = Workflow(workflow_id='simple_workflow', definition=[SimpleJob('1'), SimpleJob('2')])
-hello_world_workflow.run_job('2')
->>> '2020-01-02 01:11:00'
-```
+The `workflow_id` parameter is a string that should be a valid Python variable name. For example - `'my_example_workflow'`, `'MY_EXAMPLE_WORKFLOW'`, `'workflow1234'`.
 
-There are 2 additional parameters, that job can supply - `retry_count` and `retry_pause_sec`. The `retry_count` parameter
-determines how many times a job will be retried. The `retry_pause_sec` sets how long the pause between retries should be.
+The `Workflow` class arranges jobs into a DAG, through the `definition` parameter. There are 2 ways of specifying job arrangement. First one is passing a list
```

Comment:
DAG should be linked, maybe to Airflow DAG docs?

Reply:
done